python/src/main/python/drivers/runspv.py

```diff
@@ -0,0 +1,944 @@
+            time.sleep(BUSY_WAIT_SLEEP_SLOW)
+
+
+################################################################################
+# Legacy worker: image test
+
+
+def run_image_android_legacy(vert: str, frag: str, json: str, output_dir: str, force: bool, skip_render: bool):
+    assert (os.path.isfile(vert))
+    assert (os.path.isfile(frag))
+    assert (os.path.isfile(json))
+
+    logfile = output_dir + os.sep + LOGFILE_NAME
+    statusfile = output_dir + os.sep + 'STATUS'
+
+    prepare_device(not force, True)
+
+    adb_check('push ' + vert + ' ' + ANDROID_SDCARD_GRAPHICSFUZZ_DIR + '/test.vert.spv')
+    adb_check('push ' + frag + ' ' + ANDROID_SDCARD_GRAPHICSFUZZ_DIR + '/test.frag.spv')
+    adb_check('push ' + json + ' ' + ANDROID_SDCARD_GRAPHICSFUZZ_DIR + '/test.json')
+
+    cmd = 'shell am start -n ' + ANDROID_LEGACY_APP + '/android.app.NativeActivity'
+    flags = '--num-render {}'.format(NUM_RENDER)
+    if skip_render:
+        flags += ' --skip-render'
+
+    # Pass command line args as Intent extra. Need to nest-quote, hence the "\'flags\'"
+    cmd += ' -e gfz "\'' + flags + '\'"'
+    adb_check(cmd)
+
+    # Busy wait
+    deadline = time.time() + TIMEOUT_APP
+
+    status = 'UNEXPECTED_ERROR'
+
+    # If the app never starts, status remains as UNEXPECTED_ERROR.
+    # Once the app starts, status becomes TIMEOUT.
+    # If we reach the end of the loop below, we break and the status is updated just before.
+
+    while time.time() < deadline:
+        time.sleep(BUSY_WAIT_SLEEP_SLOW)
+
+        # Don't pass here until app has started.
+        if status == 'UNEXPECTED_ERROR':
+            if adb_can_fail('shell test -f ' + ANDROID_SDCARD_GRAPHICSFUZZ_DIR + '/STARTED').returncode != 0:
+                continue
+            status = 'TIMEOUT'
+
+        assert status == 'TIMEOUT'
+
+        # DONE file indicates app is done.
+        if adb_can_fail('shell test -f ' + ANDROID_SDCARD_GRAPHICSFUZZ_DIR + '/DONE').returncode == 0:
+            status = 'SUCCESS'
+            break
+
+        # Otherwise, keep looping/waiting while the app is still running.
+        # Quote >/dev/null otherwise this fails on Windows hosts.
+        if adb_can_fail('shell "pidof ' + ANDROID_LEGACY_APP + ' > /dev/null"').returncode == 0:
+            continue
+
+        # The app has crashed. Check for DONE file one more time.
+        if adb_can_fail('shell test -f ' + ANDROID_SDCARD_GRAPHICSFUZZ_DIR + '/DONE').returncode == 0:
+            status = 'SUCCESS'
+            break
+
+        # App has terminated and there is no DONE file; this definitely looks like a crash.
+        status = 'CRASH'
+        break
+
+    # Grab log:
+    with open(logfile, 'w', encoding='utf-8', errors='ignore') as f:
+        adb_check('logcat -d', stdout=f)
+
+    # retrieve all files
+    adb_check('pull ' + ANDROID_SDCARD_GRAPHICSFUZZ_DIR + ' ' + output_dir)
+
+    # Check sanity:
+    if status == 'SUCCESS':
+        sanity_before = output_dir + '/sanity_before.png'
```

Comment:
"/" should not be used (except on the Android device). I will try to replace these.

Reply:
I replaced these